components/src/CartCreate/cart.create.tsx

```diff
@@ -0,0 +1,456 @@
+          // eslint-disable-next-line no-console
+          console.error(error.message);
+          this.setState({ isLoading: false });
+        });
+    });
+    event.preventDefault();
+  }
+
+  renderOptions() {
+    const { productData } = this.props;
+    const productOptions = productData && productData._definition[0]._options && productData._definition[0]._options[0];
+    return (
+      <div className="product-options">
+        <h3>{productData._definition[0]['display-name']}</h3>
+        <ul className="option-list">
+          {productOptions && productOptions._element.map(option => (
+            <li className="option" key={option['display-name']}>
+              <label htmlFor="option-value" className="option-name">
+                {option['display-name']}
+                  :&nbsp;
+              </label>
+              <span className="option-value">
+                {(option._value)
+                  ? option._value[0]['display-name']
+                  : ('')}
+              </span>
+            </li>
+          ))}
+        </ul>
+      </div>
+    );
+  }
+
+  render() {
+    const { cartElements, showAddNewCartForm, isLoading } = this.state;
+    const {
+      handleModalClose, openModal, productData,
+    } = this.props;
+
+    const itemCodeString = productData._code && productData._code[0].code;
+    const modalTitle = productData._definition ? 'add-to-cart' : 'manage-carts';
+    return (
+      <Modal open={openModal} onClose={handleModalClose}>
+        <div className="modal-lg cart-create-modal">
+          <div className="modal-content">
+            <div className="modal-header">
+              <h2 className="modal-title">
+                {intl.get(modalTitle)}
+              </h2>
+            </div>
+            <div className="modal-body">
+              { productData && productData._definition ? (
+                <div className="product-data">
+                  <div className="thumbnail-col" data-el-value="lineItem.thumbnail">
+                    <Link to={`itemdetail/${encodeURIComponent(itemCodeString)}`}>
+                      <img
+                        src={Config.skuImagesUrl.replace('%sku%', itemCodeString)}
+                        onError={(e) => {
+                          const element: any = e.target;
+                          element.src = imgMissingHorizontal;
+                        }}
+                        alt={intl.get('none-available')}
+                        className="cart-lineitem-thumbnail"
+                      />
+                    </Link>
+                  </div>
+                  {this.renderOptions()}
+                </div>
+              ) : (
+                <div className="create-cart-btn-wrap">
+                  <button type="button" className="ep-btn create-cart-btn" onClick={this.handleShowCartForm}>{intl.get('create-new-cart')}</button>
+                </div>
+              )}
+              <div className="carts-list-wrap">
```

Comment:
This needs to scroll within the modal as per the mockup. It should not expand vertically as new carts are added.
![image](https://user-images.githubusercontent.com/39800563/66666298-502fe180-ec1e-11e9-8ecd-af62d129224a.png)

Reply:
**Update:** this is expected to change. I'll post context over slack.